### PR TITLE
Add ChatGPT to determine title and author

### PR DESCRIPTION
**Pull request by ederic-oytas**
# Pull Request Template

## Description

Major changes:
- Added `vision/extraction.py` which utilizes ChatGPT to extract the title and author from a set of texts recognized by the OCR model. The main function to use here is `extract_from_recognized_texts()`, which takes in a list of texts and outputs an extraction result, containing a list of possible title/author pairs.
- Added `scripts/extract.py` which takes in an image, recognizes text from that image, then uses the extraction feature to get the title and author.
- Made arguments in poe tasks replaceable. You can replace the arguments with your own arguments. This is useful if you want to test/lint/format/type-check a specific file. For example, instead of doing `poe test`, you can do `poe test tests/test_extraction.py` to run the tests specifically in that file.

Minor changes:
- The `easyocr` module is now lazily imported by placing the import inside the `TextRecognizer` class. This is done for any executions which don't use any recognition.
- Removed unneeded dependencies: `opencv-python`, `opencv-python-headless`. These are already covered by `opencv-contrib-python`, and it's recommended by the OpenCV documentation to only install one.
- Made unit tests offline (so they don't use the internet when testing) using `--disable-socket` from the `pytest-socket` plugin.

## Type of Change

- [X] Bugfix
- [X] New Feature
- [X] Documentation Update
- [ ] Other (please describe):

## Checklist

- [X] I have read the contributing guidelines.
- [X] I have followed the code style of this project.
- [X] I have added tests to cover my changes.
- [X] I have updated the documentation (if applicable).
- [X] My changes do not generate new warnings.

## Related Issue

#14

## Additional Notes

None.

### CODE REVIEW

**Review by LilAttacker — APPROVED** [review]
After having a look around, I think everything looks good.

My only real concern was if we were to run tests if it were to use your chatgpt key but you said you already addressed that directly, so I'm good with this.

File: server/tabby_server/vision/extraction.py [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+    Args:
+        recognized_texts: List of `RecognizedText` objects to extract from.
+    Returns:
+        An extraction result, which contains a list of options (pairs of
+        titles and authors).
+    """
+
+    # Create messages list to send as input
+    input_message = "\n".join(
```
Comment (David1000734) [inline server/tabby_server/vision/extraction.py:100]: An example input in the form of a comment would be helpful in visualizing how the code sends in data to the GPT model

File: server/tests/test_extraction.py [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+def test_extract_from_recognized_texts(mock_chat_completion):
+
+    array = np.array
+    int32 = np.int32
+    recognized_texts = [
+        RecognizedText(
+            text="Yhe",
+            corners=array(
+                [[1287, 1149], [1430, 1149], [1430, 1241], [1287, 1241]],
```
Comment (David1000734) [inline server/tests/test_extraction.py:44]: How are these numbers generated? 
Do they represent what boxes the model should look at to find characters?

**Review by David1000734 — APPROVED** [review]
Personally, I would like the main code to have a little more comments explain what each step is doing and why but overall, it's good.

**Pull request merged**